tutorials/rendering/compositor.rst

```diff
@@ -0,0 +1,371 @@
+be automatically freed when the shader is destructed.
+
+From this point onwards our code will run on the rendering thread.
+
+Our next step is a helper function that will recompile the shader if the user
+code was changed.
+
+.. code-block:: gdscript
+
+    # Check if our shader has changed and needs to be recompiled.
+    func _check_shader() -> bool:
+        if !rd:
+            return false
+
+        var new_shader_code : String = ""
+
+        # Check if our shader is dirty.
+        mutex.lock()
+        if shader_is_dirty:
+            new_shader_code = shader_code
+            shader_is_dirty = false
+        mutex.unlock()
+
+        # We don't have a (new) shader?
+        if new_shader_code.is_empty():
+            return pipeline.is_valid()
+
+        # Apply template.
+        new_shader_code = template_shader.replace("#COMPUTE_CODE", new_shader_code);
+
+        # Out with the old.
+        if shader.is_valid():
+            rd.free_rid(shader)
+            shader = RID()
+            pipeline = RID()
+
+        # In with the new.
+        var shader_source : RDShaderSource = RDShaderSource.new()
+        shader_source.language = RenderingDevice.SHADER_LANGUAGE_GLSL
+        shader_source.source_compute = new_shader_code
+        var shader_spirv : RDShaderSPIRV = rd.shader_compile_spirv_from_source(shader_source)
+
+        if shader_spirv.compile_error_compute != "":
+            push_error(shader_spirv.compile_error_compute)
+            push_error("In: " + new_shader_code)
+            return false
+
+        shader = rd.shader_create_from_spirv(shader_spirv)
+        if !shader.is_valid():
+            return false
+
+        pipeline = rd.compute_pipeline_create(shader)
+        if !pipeline.is_valid():
+            return false
+
+        return true
+
+At the top of this method we again use our mutex to protect accessing our
+user shader code and our is dirty flag.
+We make a local copy of the user shader code if our user shader code is dirty.
+
+If we don't have a new code fragment, we return true if we already have a
+valid pipeline.
+
+If we do have a new code fragment we embed it in our template code and then
+compile it.
+
+.. warning::
+    The code shown here compiles our new code in runtime.
+    This is great for prototyping as we can immediately see the effect
+    of the changed shader.
+
+    This prevents precompiling and caching this shader which may be an issues
+    on some platforms such as consoles.
+    Note that the demo project comes with an alternative example where
+    a ``glsl`` file contains the entire compute shader and this is used.
+    Godot is able to precompile and cache the shader with this approach.
+
+Finally we need to implement our effect callback, the rendering engine will call
+this at the right stage of rendering.
+
+.. code-block:: gdscript
+
+    # Called by the rendering thread every frame.
+    func _render_callback(p_effect_callback_type, p_render_data):
```

Comment:
```suggestion
    func _render_callback(effect_callback_type, render_data):
```
General GDScript style

Reply:
Same reasoning as on the demo repo, this will cause a name clash with class properties.